enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+> I want it to be clear when I am looking at the desired versus actual state of the system. For example, if I can see 
+> the state of the clusterversion or a machineconfigpool, it should be straightforward to understand why I am 
+> observing differences in the state of those resources compared to the state of the system."
+
+### Goals
+
+1. Indirectly support the strategic separation of control-plane and worker-node update phases for Standalone clusters by supplying a change control mechanism that will allow both control-plane and worker-node updates to proceed at predictable times without doubling operational overhead.
+2. Directly support the strategic separation of control-plane and worker-node update phases by implementing a "manual" change management strategy where users who value the full control of the separation can manually actuate changes to them independently.
+3. Empower OpenShift cluster lifecycle administrators with tools that simplify implementing industry standard notions of maintenance windows.
+4. Provide Service Delivery a platform native feature which will reduce the amount of custom tooling necessary to provide maintenance windows for customers.
+5. Deliver a consistent change management experience across all platforms and profiles (e.g. Standalone, ROSA, HCP).
+6. Enable SRE to, when appropriate, make configuration changes on a customer cluster and have that change actually take effect only when permitted by the customer's change management preferences.
+7. Do not subvert expectations of customers well served by the existing fully self-managed cluster update.
+8. Ensure the architectural space for enabling different change management strategies in the future. 
+
+### Non-Goals
+
+1. Allowing control-plane upgrades to be paused midway through an update. Control-plane updates are relatively rapid and pausing will introduce unnecessary complexity and risk. 
+2. Requiring the use of maintenance schedules for OpenShift upgrades (the changes should be compatible with various upgrade methodologies – including being manually triggered).
+3. Allowing Standalone worker-nodes to upgrade to a different payload version than the control-plane (this is supported in HCP, but is not a goal for standalone).
+4. Exposing maintenance schedule controls from the oc CLI. This may be a future goal but is not required by this enhancement.
+5. Providing strict promises around the exact timing of upgrade processes. Maintenance schedules will be honored to a reasonable extent (e.g. upgrade actions will only be initiated during a window), but long-running operations may exceed the configured end of a maintenance schedule.
+6. Implementing logic to defend against impractical maintenance schedules (e.g. if a customer configures a 1-second maintenance schedule every year). Service Delivery may want to implement such logic to ensure upgrade progress can be made.
```

Comment:
If we (SD) need to put something in place to manage this it's going to mean keeping some code somewhere.  Today scheduling is via OCM.  If this enhancement is accepted this becomes a platform capbility.  We (SD) have not talked about the implications of such a feature on how we enable customer to manage upgrades.  Can we see a world where SD gets out of the business of managing schedules completely at some point in the future?  If so, having the right controls in place to ensure sane configurations will generally be a good idea.  

That being said, we only care about the version of the cluster in a few cases.
1. if the cluster version is end of life (EOL)
2. if the cluster's control plane is running on RH SRE infra

In the #1 scenario, the cluster goes into Limited Support.
In the #2 scenario, it only applys for HCP toploogies where the pods of the control plane are running on RH infrastructure.  SRE will force a control plane upgrade if the control plane version for signifigent bugs that need remediated, there is a vulnerability we need to remediate, or the control plane version is now EOL.

Reply:
> Can we see a world where SD gets out of the business of managing schedules completely at some point in the future?

Yes. SD could let customers fully define their own maintenance schedules and simply override them when necessary (e.g. using `disabledUntil`). The reason SD _might_ want to play a larger role in constraining the setting is to make sure, when necessary, that forced updates are performed within a customer's desired window.  Your #2 scenario, for example: you have to do the update and ideally want users to be OK with when it happens. If you force them to choose a non-zero change management window each month, hopefully they are less surprised when forced changes happen during that window.